### PR TITLE
test: added key press tests

### PR DESCRIPTION
**Pull request by joannalikespi**
added in tests for each of the key press commands on the recording page to prove that the new key handlers work

### CODE REVIEW

File: src/renderer/components/recording-page/recording-page.test.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/**
```
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: These shouldn't be necessary.

File: src/renderer/components/recording-page/recording-page.test.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import RecordingPage from './index';
+
+// Mock the remote module to prevent import failure.
+jest.mock('@electron/remote', () => ({ dialog: jest.fn() }));
+
+jest.mock('../../utils/media-utils');
+
+// Automatic mock doesn't work because the import in the module causes error when in a testing environment
+jest.mock('../../services/media', () => ({
+  switchOnAudioInput: jest.fn(noOp),
```
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: You don't need to pass `noOp` if is just a no-op

File: src/renderer/components/recording-page/recording-page.test.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  playBlob: jest.fn(noOp),
+  createOscillator: jest.fn(noOp),
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  playAudioNode: jest.fn((node: AudioScheduledSourceNode | AudioNode) => {
+    return new Promise(noOp);
+  }),
+  stopPlaying: jest.fn(noOp),
+}));
+
+jest.mock;
```
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: Clean up

File: src/renderer/components/recording-page/recording-page.test.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+describe('recordingPage', () => {
```
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: `RecordingPage`

File: src/renderer/components/recording-page/recording-page.test.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  it('starts recording when r is pressed', async () => {
+    expect.assertions(4);
+
+    // Fix for 'Reference Error: __static is not defined
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (global as any).__static = '';
```
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: This should be fixed in `jest.setup.js` or in `jest.config.js` as extra globals

File: src/renderer/components/recording-page/recording-page.test.tsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+    //Define mocks to confirm that calls have been made
```
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: Space after `//`

File: src/renderer/components/recording-page/recording-page.test.tsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    const alertStartRecording = jest.spyOn(mediaService, 'startRecording');
+    const alertAudioBlobToWavArrayBuffer = jest.spyOn(mediaService, 'audioBlobToWavArrayBuffer');
+    const stopRecording = jest.spyOn(mediaService, 'stopRecording');
+    const switchOnAudio = jest.spyOn(mediaService, 'switchOnAudioInput');
+
+    render(<RecordingPage onBack={noOp} recordingItems={[]} basePath={''} scaleKey={'C'} octave={2} />);
+    //R is 82
```
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: Space after `//`

File: src/renderer/components/recording-page/recording-page.test.tsx [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+    const osc = audioContext.createOscillator();
+
+    //Define mocks to confirm that calls have been made
+    const createNewAudioGraph = jest.spyOn(mediaService, 'createNewAudioGraph');
+    const createOscillator = mocked(mediaService.createOscillator);
+    osc.start();
+    createOscillator.mockReturnValue(osc);
+
+    const playAudioNode = mocked(mediaService.playAudioNode);
+    playAudioNode.mockResolvedValue(new Promise(noOp));
```
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: What is this trying to accomplish? A `Promise` that doesn't call `resolve` is a broken Promise that never resolves.

**Review by iluminar-yi — CHANGES_REQUESTED** [review]
Update PR title to follow the commit message convention

File: src/renderer/components/recording-page/recording-page.test.tsx (hunk 8 above)
Comment (joannalikespi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: I was trying to mock the method to return a Promise where resolve didn't do anything - am I going about it the wrong way?

**Comment by joannalikespi** [comment]
> Update PR title to follow the commit message convention

Is "build" an appropriate prefix? I wasn't sure since it's adding tests - "feat" and "fix" didn't feel right.

File: jest.setup.js [hunk 9]
```diff
@@ -8,6 +8,9 @@
   global.TextDecoder = TextDecoder;
 }
 
+// Fix for 'Reference Error: __static is not defined
+global.__static = '';
```
Comment (iluminar-yi) [inline jest.setup.js]: Use `__static` for this value

File: src/renderer/components/recording-page/recording-page.test.tsx (hunk 8 above)
Comment (iluminar-yi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: `playAudioNode()` returns `Promise<void>` which means its resolved value is a `void` - here you are saying the resolved value is a `Promise`. That's disconnect No. 1.

`Promise<void>` should be `new Promise(resolve => resolve())`. Passing `noOp` means `resolve` will never be called, so the Promise will never resolve. That's disconnect No. 2.

**Review by iluminar-yi — CHANGES_REQUESTED** [review]
Please rebase this branch.

File: jest.setup.js (hunk 9 above)
Comment (joannalikespi) [reply jest.setup.js]: I wasn't sure if you meant to change the left or right side of the assignment, but I just tried both 
`__static = '';` and `global.__static=__static;` and both generate the reference error that `global.__static=''` fixes

**Review by iluminar-yi — APPROVED** [review]

**Pull request merged**